src/components/Dropdown/Dropdown.jsx

```diff
@@ -0,0 +1,81 @@
+import PropTypes from "prop-types";
+import React, { useRef, useEffect } from "react";
+import styles from './styles.css';
+import Icon from '../Icon/Icon.jsx';
+
+const Dropdown = ({
+  children,
+  title,
+  handleDropdown,
+  show,
+  style,
+  onSelectItem,
+  closeOutsideClick,
+  ...props
+}) => {
+	const renderChildrenItems = () => {
+    return React.Children.map(children, (child, index) => {
+      return React.cloneElement(child, {
+      	style: style,
+      	onSelect: onSelectItem, 
+        itemindex: index,
+      });
+    });
+  };
+
+	function useOutsideAlerter(ref) {
```

Comment:
You could turn the `useOutsideAlerter` it into a reusable hook for example: `useOnClickOutside`. It would be pretty similar to the `useHover` hook.
It would be something like this:

```js
export default function useClickOutside(onClickHandler) {
  // Wrap in useCallback so we can use in dependencies below
  const handleClick = useCallback(() => onClickHandler(), []);

  // Keep track of the last node passed to callbackRef
  // so we can remove its event listeners.
  const ref = useRef();
  const callbackRef = useCallback(
    (node) => {
      if (ref.current) {
        ref.current.removeEventListener("mousedown", handleClick);
      }

      ref.current = node;

      if (ref.current) {
        ref.current.addEventListener("mousedown", handleClick);
      }
    },
    [handleClick]
  );

  return [callbackRef];
}

```

I haven't test that but just to give you a better idea